ddtrace/internal/writer.py

```diff
@@ -382,6 +383,10 @@ def _set_keep_rate(self, trace):
         if trace:
             trace[0].set_metric(KEEP_SPANS_RATE_KEY, 1.0 - self._drop_sma.get())
 
+    def _set_git_metadata(self, trace):
+        if trace:
+            trace[0].set_tags(gitmetadata.get_tracer_tags())
```

Comment:
We generally avoid adding trace level tags in the agent writer. We should move `trace[0].set_tags(gitmetadata.get_tracer_tags())` to the [TraceTagsProcessor](https://github.com/DataDog/dd-trace-py/blob/v1.10.1/ddtrace/internal/processor/trace.py#L129).

Reply:
We should inject it in the first span of each payload (not root span)
Is it possible int the [TraceTagsProcessor](https://github.com/DataDog/dd-trace-py/blob/v1.10.1/ddtrace/internal/processor/trace.py#L129)?

Reply:
Yes, [TraceTagsProcessor](https://github.com/DataDog/dd-trace-py/blob/v1.10.1/ddtrace/internal/processor/trace.py#L129) adds tags to the first span in a payload not just the root span.